src/net/neighbors/db.rs

```diff
@@ -0,0 +1,652 @@
+        PeerDB::asn_count(network.peerdb_conn(), asn).unwrap_or(1)
+    }
+}
+
+impl NeighborWalkDB for PeerDBNeighborWalk {
+    fn get_fresh_random_neighbors(
+        &self,
+        network: &PeerNetwork,
+        num_neighbors: u64,
+    ) -> Result<Vec<Neighbor>, net_error> {
+        Self::inner_get_fresh_random_neighbors(network, num_neighbors)
+    }
+
+    fn get_initial_walk_neighbors(
+        &self,
+        network: &PeerNetwork,
+        ibd: bool,
+    ) -> Result<Vec<Neighbor>, net_error> {
+        Self::inner_get_initial_walk_neighbors(network, ibd)
+    }
+
+    fn lookup_stale_neighbors(
+        &self,
+        network: &PeerNetwork,
+        addrs: &Vec<NeighborAddress>,
+    ) -> Result<(HashMap<NeighborAddress, Neighbor>, Vec<NeighborAddress>), net_error> {
+        Self::inner_lookup_stale_neighbors(network, addrs)
+    }
+
+    fn add_or_schedule_replace_neighbor(
+        &self,
+        network: &mut PeerNetwork,
+        naddr: &NeighborAddress,
+        preamble: &Preamble,
+        handshake: &HandshakeData,
+        db_data: Option<&StackerDBHandshakeData>,
+        replacements: &mut NeighborReplacements,
+    ) -> Result<(bool, Neighbor), net_error> {
+        Self::inner_add_or_schedule_replace_neighbor(
+            network,
+            naddr,
+            preamble,
+            handshake,
+            db_data,
+            replacements,
+        )
+    }
+
+    fn check_neighbor_denied(
+        &self,
+        network: &PeerNetwork,
+        nk: &NeighborKey,
+    ) -> Result<(), net_error> {
+        Self::inner_check_neighbor_denied(network, nk)
+    }
+
+    fn replace_neighbors(
+        &self,
+        network: &mut PeerNetwork,
+        replacements: &NeighborReplacements,
+        result: &mut NeighborWalkResult,
+    ) -> Result<(), net_error> {
+        Self::inner_replace_neighbors(network, replacements, result)
+    }
+
+    fn neighbor_from_handshake(
+        &self,
+        network: &PeerNetwork,
+        preamble: &Preamble,
+        data: &HandshakeAcceptData,
+    ) -> Result<Neighbor, net_error> {
+        Self::inner_neighbor_from_handshake(network, preamble, data)
+    }
+
+    fn save_neighbor_from_handshake(
+        &self,
+        network: &mut PeerNetwork,
+        preamble: &Preamble,
+        data: &HandshakeAcceptData,
+        db_data: Option<&StackerDBHandshakeData>,
+    ) -> Result<Neighbor, net_error> {
+        Self::inner_save_neighbor_from_handshake(network, preamble, data, db_data)
+    }
+
+    fn update_neighbor(
+        &self,
+        network: &mut PeerNetwork,
+        cur_neighbor: Neighbor,
+        new_data: Option<&HandshakeAcceptData>,
+        new_db_data: Option<&StackerDBHandshakeData>,
+    ) -> Result<Neighbor, net_error> {
+        Self::inner_update_neighbor(network, cur_neighbor, new_data, new_db_data)
+    }
+
+    fn get_asn_count(&self, network: &PeerNetwork, asn: u32) -> u64 {
+        Self::inner_get_asn_count(network, asn)
+    }
```

Comment:
Why add all these one-line shuttle invocations of `inner_*`? It seems like those methods could all just be implemented right in the trait implementation.

Reply:
Sure, I can do this.  I just like having my implementation separate from the trait binding, but that's really not needed here.